src/pages/index.js

```diff
@@ -63,13 +107,81 @@ const cardList = new Section(
 // Card create function
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", (data) => {
-    imagePopup.open(data);
-  });
+  const card = new Card(
+    cardData,
+    "#card-template",
+    (data) => {
+      imagePopup.open(data);
+    },
+    handleDeleteCardClick,
+    handleLikeClick
+  );
   return card.getView();
 }
 
-//
+// Like Button
+
+function handleLikeClick(cardID, cardInstance) {
+  if (cardInstance._isLiked) {
+    api
+      .removeLike(cardID)
+      .then(() => {
+        cardInstance.setCardLike(false);
+        console.log("like removed for card ID", cardID);
+      })
+      .catch((error) => console.error("error removing like:", error));
+  } else {
+    api
+      .addLike(cardID)
+      .then(() => {
+        cardInstance.setCardLike(true);
+      })
+      .catch((error) => console.error("Error adding like", error));
+  }
+}
+
+// Delete Button
+
+function handleDeleteCardClick(cardInstance) {
+  confirmPopup.open();
+  confirmPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardInstance._id)
+      .then(() => {
+        cardInstance.handleDeleteCard();
+        confirmPopup.close();
+      })
+      .catch((error) => console.error("Error deleting cards:", error));
+  });
+}
+
+// Update Avatar
+
+avatarEditBtn.addEventListener("click", () => {
+  updateAvatarPopup.open();
+});
+
+const updateAvatarPopup = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: (userData) => {
+    updateAvatarPopup.setIsLoading(true);
+    const avatarURL = userData.link;
+    api
+      .updateAvatar(avatarURL)
+      .then((updatedUserData) => {
+        avatar.src = updatedUserData.avatar;
+        updateAvatarPopup.close();
+        updateAvatarPopup.resetForm();
+        avatarFormValidator.disableSubmitButton();
+      })
+      .catch((err) => console.error("Error updating avatar", err))
+      .finally(() => {
+        updateAvatarPopup.setIsLoading(false);
```

Comment:
* I can't change the avatar. See the error https://snipboard.io/FQh0sy.jpg 

`renderLoading` should be used here